inc/ticket.class.php

```diff
@@ -6838,4 +6764,269 @@ public static function buildCanViewCondition($fieldID) {
+                  array_filter($tomerge, function($key) use ($tomerge_ids) {
+                     return in_array($key, $tomerge_ids);
+                  });
+
+                  foreach ($tomerge as $document_item2) {
+                     $document_item2['items_id'] = $merge_target_id;
+                     unset($document_item2['id']);
+                     if (!$document_item->add($document_item2)) {
+                        //Cannot add followup. Abort/fail the merge
+                        throw new \RuntimeException(sprintf(__('Failed to add document to ticket %d'), $merge_target_id), 1);
+                     }
+                  }
+               }
+               if ($p['link_type'] > 0 && $p['link_type'] < 5) {
+                  //Add relation (this is parent of merge target)
+                  $tt = new Ticket_Ticket();
+                  $linkparams = [
+                     'link'         => $p['link_type'],
+                     'tickets_id_1' => $id,
+                     'tickets_id_2' => $merge_target_id
+                  ];
+                  $tt->deleteByCriteria([
+                     'OR' => [
+                        'AND' => [
+                           'tickets_id_1' => $merge_target_id,
+                           'tickets_id_2' => $id
+                        ],
+                        'AND' => [
+                           'tickets_id_2' => $merge_target_id,
+                           'tickets_id_1' => $id
+                        ]
+                     ]
+                  ]);
+                  if (!$tt->add($linkparams)) {
+                     //Cannot link tickets. Abort/fail the merge
+                     throw new \RuntimeException(sprintf(__('Failed to link tickets %d and %d'), $merge_target_id, $id), 1);
+                  }
+               }
+               if (isset($p['append_actors'])) {
+                  $tu = new Ticket_User();
+                  $existing_users = $tu->find(['tickets_id' => $merge_target_id]);
+                  $gt = new Group_Ticket();
+                  $existing_groups = $tu->find(['tickets_id' => $merge_target_id]);
+                  $st = new Supplier_Ticket();
+                  $existing_suppliers = $tu->find(['tickets_id' => $merge_target_id]);
+
+                  foreach ($p['append_actors'] as $actor_type) {
+                     $users = $tu->find([
+                        'tickets_id' => $id,
+                        'type' => $actor_type
+                     ]);
+                     $groups = $gt->find([
+                        'tickets_id' => $id,
+                        'type' => $actor_type
+                     ]);
+                     $suppliers = $st->find([
+                        'tickets_id' => $id,
+                        'type' => $actor_type
+                     ]);
+                     $users = array_filter($users, function($user) use ($existing_users) {
+                        foreach ($existing_users as $existing_user) {
```

Comment:
You have to handle case of external users. Indeed, you can for example add an observer with only an email address. Its `users_id` will be 0 and the only comparable information in this case would be its `alternative_email` field.